library/std/src/sys/unix/thread_local_dtor.rs

```diff
@@ -12,6 +12,9 @@
 // compiling from a newer linux to an older linux, so we also have a
 // fallback implementation to use as well.
 #[cfg(any(target_os = "linux", target_os = "fuchsia", target_os = "redox"))]
+// FIXME: Transmuting __cxa_thread_atexit_impl and indirectly branching
+// to/calling it here is an explicit CFI violation.
+#[no_sanitize(cfi)]
```

Comment:
What exactly is the CFI violation, here? If the concern is simply that `*mut u8` is not `*mut c_void`, then it should work if it casts to `*mut c_void`, no?

Though... why does CFI care about the difference between `*mut c_void` and `*mut u8`?

Reply:
(yeah, especially since `c_void` is `repr(u8)`)

Reply:
I just adjusted the casts/transmutes in this indirect call site (see https://github.com/rust-lang/rust/issues/115199#issuecomment-1693957442). Let me know if it looks good to you. I'll take a look at the other two.

Reply:
(I'll also take a look at the CI errors.)

Reply:
Okay, excellent. I'm still concerned about the need to change this in the general sense, but this looks much more understandable.